### ISSUE TITLE
[fix] nav buttons ahould always have aria-label attributes

### ISSUE THREAD

**Issue by seevee** [trussworks/react-uswds#2639]
**ReactUSWDS Version & USWDS Version:**

* ReactUSWDS: 5.5.0
* USWDS: 3.6.0

**Describe the bug**

[WCAG § 4.1.2](https://www.w3.org/TR/WCAG21/#name-role-value) states the following:

> For all [user interface components](https://www.w3.org/TR/WCAG21/#dfn-user-interface-components) (including but not limited to: form elements, links and components generated by scripts), the [name](https://www.w3.org/TR/WCAG21/#dfn-name) and [role](https://www.w3.org/TR/WCAG21/#dfn-role) can be [programmatically determined](https://www.w3.org/TR/WCAG21/#dfn-programmatically-determinable); states, properties, and values that can be set by the user can be [programmatically set](https://www.w3.org/TR/WCAG21/#dfn-programmatically-set); and notification of changes to these items is available to [user agents](https://www.w3.org/TR/WCAG21/#dfn-user-agents), including [assistive technologies](https://www.w3.org/TR/WCAG21/#dfn-assistive-technologies).

The `NavCloseButton` element, while it accepts an `aria-label` itself, by default only applies an `aria-label` to the enclosed `Icon.Close` rather than the `<button />` element. This means that using any further abstracted components (like `ExtendedNav`) will cause the resulting HTML to violate this rule.

The buttons within the modal elements already conform to this paradigm, only the nav elements appear to be effected.

**To Reproduce**

1. Go to the project storybook
2. Click on Components -> Header
3. Inspect any `*Button` element
4. Observe that `aria-label` is not present on button elements, but is present on child `<svg />` elements

**Expected behavior**

`<button />` elements should have `aria-label` attribute to meet WCAG § 4.1.2 requirements.

**Additional context**

This issue was initially identified using [HTML_Codesniffer](https://github.com/squizlabs/HTML_CodeSniffer) which we use alongside Axe and playwright to implement E2E testing.
